packages/react-events/src/Swipe.js

```diff
@@ -0,0 +1,217 @@
+          }
+          const x = (obj: any).screenX;
+          const y = (obj: any).screenY;
+
+          let shouldEnableSwiping = true;
+
+          if (props.onShouldClaimOwnership && props.onShouldClaimOwnership()) {
+            shouldEnableSwiping = context.requestOwnership(eventTarget);
+          }
+          if (shouldEnableSwiping) {
+            state.isSwiping = true;
+            state.startX = x;
+            state.startY = y;
+            state.x = x;
+            state.y = y;
+            state.swipeTarget = eventTarget;
+            context.addRootEventTypes(rootEventTypes);
+          } else {
+            state.touchId = null;
+          }
+        }
+        break;
+      }
+      case 'touchmove':
+      case 'mousemove':
+      case 'pointermove': {
+        if (context.isPassive()) {
+          return;
+        }
+        if (state.isSwiping) {
+          let obj = null;
+          if (eventType === 'touchmove') {
+            const targetTouches = (event: any).targetTouches;
+            for (let i = 0; i < targetTouches.length; i++) {
+              if (state.touchId === targetTouches[i].identifier) {
+                obj = targetTouches[i];
+                break;
+              }
+            }
+          } else {
+            obj = event;
+          }
+          if (obj === null) {
+            state.isSwiping = false;
+            state.swipeTarget = null;
+            state.touchId = null;
+            context.removeRootEventTypes(rootEventTypes);
+            return;
+          }
+          const x = (obj: any).screenX;
+          const y = (obj: any).screenY;
+          if (x < state.x && props.onSwipeLeft) {
+            state.direction = 3;
+          } else if (x > state.x && props.onSwipeRight) {
+            state.direction = 1;
+          }
+          state.x = x;
+          state.y = y;
+          if (props.onSwipeMove) {
+            const eventData = {
+              diffX: x - state.startX,
+              diffY: y - state.startY,
+            };
+            dispatchSwipeEvent(
+              context,
+              'swipemove',
+              props.onSwipeMove,
+              state,
+              false,
+              eventData,
+            );
+            (event: any).preventDefault();
+          }
+        }
+        break;
+      }
+      case 'pointercancel':
+      case 'touchcancel':
+      case 'touchend':
+      case 'mouseup':
+      case 'pointerup': {
+        if (state.isSwiping) {
+          if (state.x === state.startX && state.y === state.startY) {
+            return;
+          }
+          if (props.onShouldClaimOwnership) {
+            context.releaseOwnership(state.swipeTarget);
+          }
+          const direction = state.direction;
+          const lastDirection = state.lastDirection;
+          if (direction !== lastDirection) {
+            if (props.onSwipeLeft && direction === 3) {
+              dispatchSwipeEvent(
+                context,
+                'swipeleft',
```

Comment:
I think it might be preferable to have a single event like `onSwipe` with event data that provides the direction, velocity, etc. That would be simpler to work with in userland than potentially listening to 4 events.